### ISSUE TITLE
Incorrect Source Address select

### ISSUE THREAD

**Issue by ZhiShengYuan** [AdguardTeam/AdGuardHome#6527]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, AMD64 (aka x86_64)

### Installation

GitHub releases or script from README

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

  v0.107.42

### Action

```sh
dig @192.168.5.1  www.google.com
```

### Expected result

it works well

### Actual result

AdguardHome, when responding to DNS requests, uses the interface address of the best routing interface instead of the target address of the client.

### Additional information and/or screenshots

I have a device with multiple interfaces. The `eno2` interface has the address `192.168.5.1` , and the `eno4` interface has the address `10.38.0.253` . When I attempt to use the client with the IP `10.38.0.184` to utilize `192.168.5.1` as the DNS server, AdGuard responds with the source address `10.38.0.253` instead of `192.168.5.1` .

The issue is not related to nftables, other firewalls, or NAT. This is confirmed because the functionality operates correctly in dnsmasq, where dnsmasq uses the correct source address.

And the adguardhome is bind to `0.0.0.0`

here is the tcpdump result
AdGuardHome : 
```
21:50:29.418893 eno2  In  IP 10.38.0.184.47496 > 192.168.5.1.55: UDP, length 51
21:50:29.419431 eno4  Out IP 10.38.0.253.55 > 10.38.0.184.47496: UDP, length 119
21:50:34.425104 eno2  In  IP 10.38.0.184.38633 > 192.168.5.1.55: UDP, length 51
21:50:34.425639 eno4  Out IP 10.38.0.253.55 > 10.38.0.184.38633: UDP, length 119
21:50:39.431504 eno2  In  IP 10.38.0.184.38528 > 192.168.5.1.55: UDP, length 51
21:50:39.432124 eno4  Out IP 10.38.0.253.55 > 10.38.0.184.38528: UDP, length 119
```

dnsmasq:
```
22:01:05.660238 eno2  In  IP 10.38.0.184.58983 > 192.168.5.1.53: 11254+ [1au] A? www.google.com. (55)
22:01:05.661193 eno4  Out IP 192.168.5.1.53 > 10.38.0.184.58983: 11254 1/0/1 A 216.58.200.228 (59)
```

**Comment by ZhiShengYuan** [comment]
btw, here is the routing table 
```
10.38.0.0/24 dev eno4 proto kernel scope link src 10.38.0.253 metric 999
10.38.0.0/16 via 192.168.5.254 dev eno2 proto static metric 200
```

**Comment by ZhiShengYuan** [comment]
dump to https://github.com/AdguardTeam/AdGuardHome/issues/3015

**Comment by hanjo** [comment]
As I wrote in the other issue, I'm also experiencing this in [v0.107.42](https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.42), while v0.107.41 was working fine.

**Comment by ZhiShengYuan** [comment]
> As I wrote in the other issue, I'm also experiencing this in [v0.107.42](https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.42), while v0.107.41 was working fine.

it seems that fixed in v0.107.43
